From ceab990c13bdbab073189957d0a6e44d5b22cdbe Mon Sep 17 00:00:00 2001
From: tonishi92 <tatsuo.onishi@gmail.com>
Date: Mon, 9 Jan 2023 09:16:42 +0100
Subject: [PATCH 71/71] (Debug Update) Update to a previous debug. "Keep p and
 th inside the bound  IPTB and ITHTB are kept within [1,220] and [1,250],
 respectively"

The same debug was applied in a previous commit.
In this commit, the same update is applied to another place in the same file.
Needs to be merged with the previous commit.

List of modified files:
   * phys/module_cu_kfcup.F
---
 phys/module_cu_kfcup.F | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/phys/module_cu_kfcup.F b/phys/module_cu_kfcup.F
index 0f0e802..df276fa 100644
--- a/phys/module_cu_kfcup.F
+++ b/phys/module_cu_kfcup.F
@@ -4120,7 +4120,15 @@ iter:     DO NCOUNT=1,10
       tth=(thes-bth)*rdthk
       pp   =tth-aint(tth)
       ithtb=int(tth)+1
-!
+!TO2
+      IF(IPTB.GE.220 .OR. IPTB.LE.1 .OR. ITHTB.GE.250 .OR. ITHTB.LE.1)THEN
+        write(*,*) '**** OUT OF BOUNDS ********** IPTB and ITHTB'
+        IPTB = max(1,iptb)
+        IPTB = min(219,iptb)
+        ithtb = max(1,ithtb)
+        ithtb = min(249,ithtb)
+      ENDIF
+!TO2
       t00=ttab(ithtb  ,iptb  )
       t10=ttab(ithtb+1,iptb  )
       t01=ttab(ithtb  ,iptb+1)
-- 
2.31.1

